File: c++/Evaluate_Reverse_Polish_Notation.cpp
class Solution {
public:
    int evalRPN(vector<string> &tokens) {
        if(tokens.empty())
        {
            return 0;
        }
        stack<int> storage;
        for(int i=0;i<tokens.size();i++)
        {
            if (tokens[i]=="+")
            {
                int a=storage.top();
                storage.pop();
                storage.top()=a+storage.top();
            }
            else if(tokens[i]=="-")
            {   
                int a=storage.top();
                storage.pop();
                storage.top()=storage.top()-a;
            }
            else if(tokens[i]=="*")
            {
                int a=storage.top();
                storage.pop();
                storage.top()=storage.top()*a;
            }
            else if(tokens[i]=="/")
            {
                int a=storage.top();
                storage.pop();
                storage.top()=storage.top()/a;
            }
            else
            {
                storage.push(atoi(tokens[i].c_str()));
            }
        }
        return storage.top();
    }
};

File: python/Evaluate_Reverse_Polish_Notation.py
class Solution:
    # @param tokens, a list of string
    # @return an integer
    def evalRPN(self, tokens):
        stack=[]
        for i in tokens:
            if i=="+":
                op2=stack.pop()
                op1=stack.pop()
                stack.append(op1+op2)
            elif i=="-":
                op2=stack.pop()
                op1=stack.pop()
                stack.append(op1-op2)
            elif i=="*":
                op2=stack.pop()
                op1=stack.pop()
                stack.append(op1*op2)
            elif i=="/":
                op2=stack.pop()
                op1=stack.pop()
                stack.append(int(op1*1.0/op2)) #Python's divsion is different from c++
            else:
                stack.append(int(i))
        return stack.pop()